webapp/graphite/render/grammar_unsafe.py

```diff
@@ -80,7 +80,11 @@
 
 
 def setRaw(s, loc, toks):
-    toks[0].raw = s[toks[0].start:toks[0].end]
+    try:
```

Comment:
The code you have in the try-block should be compatible with pyparsing pre-3.0 and post-3.0. It's the code in the except block that only works in pyparsing pre 3.0.

I don't think the code in the try block will *ever* raise `AttributeError`, in either version of pyparsing.

Reply:
My bad, you're right. I initially had the try/except blocks the other way around, but swapped them to make the post-3.0 compatible change the default and forgot to update the exception handling. I hadn't checked that they both would work post and pre 3.0, and so I fixed this. Thanks for catching this.